### ISSUE TITLE
False-positive `unneeded_concatenation_linter` for `c(name = "value")`

### ISSUE THREAD

**Issue by jabenninghoff** [r-lib/lintr#1301]
The following code is generating a false positive for  `unneeded_concatenation_linter`. Appears to be introduced by #1271.

test.R:

```r
dput(c(name = "value"))
```

Linting the file:

```sh
$ Rscript test.R 
c(name = "value")

$ Rscript -e 'lintr::lint("test.R", linters = lintr::unneeded_concatenation_linter())'
test.R:1:6: style: Unneeded concatenation of a constant. Remove the "c" call.
dput(c(name = "value"))
     ^
```
